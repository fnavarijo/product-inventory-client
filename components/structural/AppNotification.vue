<template>
  <div class="bg-red-500 text-white flex items-center rounded p-2">
    <WarningIcon class="w-8 h-8 mr-4" />
    <span>{{ message }}</span>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

import WarningIcon from '@/assets/img/warning.svg?inline';

export default Vue.extend({
  name: 'AppNotification',
  components: {
    WarningIcon,
  },
  props: {
    message: {
      type: String,
      required: true,
    },
  },
});
</script>
