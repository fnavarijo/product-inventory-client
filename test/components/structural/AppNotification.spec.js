import { mount } from '@vue/test-utils';

import AppNotification from '@/components/structural/AppNotification.vue';

describe('AppNotification component', () => {
  it('should display notification with message', () => {
    const wrapper = mount(AppNotification, {
      propsData: {
        message: 'Something happened',
      },
    });

    expect(wrapper.text()).toBe('Something happened');
    expect(wrapper.find('svg').exists()).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
