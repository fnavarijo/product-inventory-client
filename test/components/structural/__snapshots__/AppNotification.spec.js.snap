// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNotification component should display notification with message 1`] = `<div class="bg-red-500 text-white flex items-center rounded p-2"><svg class="w-8 h-8 mr-4"></svg> <span>Something happened</span></div>`;
